lib/r2d2/payment_token.rb

```diff
@@ -41,11 +42,11 @@ def generate_shared_secret(private_key, ephemeral_public_key)
       end
 
       def derive_hkdf_keys(ephemeral_public_key, shared_secret)
-        key_material = Base64.decode64(ephemeral_public_key) + shared_secret;
-        hkdf = HKDF.new(key_material, :algorithm => 'SHA256', :info => 'Android')
-        hkdf_keys = {
-          :symmetric_encryption_key => hkdf.next_bytes(16),
-          :mac_key => hkdf.next_bytes(16)
+        key_material = Base64.decode64(ephemeral_public_key) + shared_secret
+        hkdf = HKDF.new(key_material, algorithm: 'SHA256', info: 'Android', salt: SALT).next_bytes(32)
+        {
+          symmetric_encryption_key: hkdf[0, 16],
+          mac_key: hkdf[16, 16]
```

Comment:
Explored this in https://github.com/bdewater/r2d2/commit/42962e4fbca133b30edc5e69cc854280e28e8a8e

Reply:
It seems the value of this change is limited given we're not also providing a way for the user to choose their HKDF implementation?

Reply:
In it's current form, true. Wouldn't be too hard to feature detect the OpenSSL implementation and fall back to the Ruby gem if we feel like it.